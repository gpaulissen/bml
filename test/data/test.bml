// BEGIN_SRC #1 in README.org
1C  Any hand with 16+ hcp
  1D  Artificial. 0--7 hcp
    1H  Any hand with 20+ hcp
  1HS 5+ suit, forcing to game (8+ hcp)
  1N  Natural game force, 8+ hcp
  2m  5+ suit, forcing to game (8+ hcp)
  2M  6+ suit, 5--7 hcp
1D  Nebulous with 2+!d, 11--15 hcp
1HS 5+ suit, 11--15 hcp
1N  14--16 hcp
2C  6+!c or 5!c and 4!h/!s, 11--15 hcp
2D  4414. 4405, 4305 or 3405, 11--15 hcp
2HS Weak
2N  Weak with 5-5 minors
 
// BEGIN_SRC #2 in README.org
1C  Polish club, one of three:
    a) 12--14 NT
    b) 15+ hcp and 5+!c
    c) 18+ hcp any distribution

// BEGIN_SRC #3 in README.org
1m = 3+ minor
1M = 5+ major
1N = 15--17
     May have 5M, 6m or 5-4 minors
2C = Strong and forcing
2X = Weak
2N = 20--21

// BEGIN_SRC #4 in README.org
1N-2C;
2D No 4 card major
  2H 5+!h, 4!s, invitational
  2S 5+!s, invitational
  3HS Smolen (5+ cards in other major)
2HS 4+ suit
2N  4-4 majors, minimum
3C  4-4 majors, maximum

// BEGIN_SRC #5 in README.org
2C Strong, forcing
  2D Waiting
2X Weak
  2N Ogust

// BEGIN_SRC #6 in README.org
(1NT)---
D   Strength, ca 15+
2C  At least 5-4 majors
  (D)
    P  5+!c, suggestion to play
    R  Asking for better/longer major
    2D 5+!d, suggestion to play
  (P)
    2D Asking for better/longer major
2D  A weak major or a strong minor
2HS Constructive
2N  5-5 minors
3X  Preemptive

// BEGIN_SRC #7 in README.org
#CUT transfer
2\R Transfer
  2\M Transfer accept
  3\M Super accept
#ENDCUT

1N---
2C Stayman
#PASTE transfer \R=D \M=H
#PASTE transfer \R=H \M=S

// BEGIN_SRC #8 in README.org
* The 1!c opening

  Opening 1!c shows at least 16+ hcp, and is forcing. The
  continuation is fairly natural.

  Some hands might be upgraded to 1!c due to distribution, but
  wildly distributional hands might also be downgraded to avoid
  problems if the opponents preempt.

** The 1!d negative

   Responding 1!d shows a hand which doesn't have enough values to
   establish a game force.

// BEGIN_SRC #9 in README.org
#+TITLE: Precision club
#+AUTHOR: John Smith
#+DESCRIPTION: Strong club system. Nebulous diamond and 5 card majors

// BEGIN_SRC #10 in README.org
N None 35 4SXS hK
N Kxx T9 xxx Q987x
E Jx AJxxx AJx KTx
S A987xx - Q9xx Axx
W QT KQxxxx KTx Jx
 
// BEGIN_SRC #11 in README.org
#35, Dealer N, None vul, played 4SXS with hK lead
N: Kxx    T9     xxx  Q987x
E: Jx     AJxxx  AJx  KTx
S: A987xx -      Q9xx Axx
W: QT     KQxxxx KTx  Jx

// BEGIN_SRC #12 in README.org
- The first item in an unordered list
- The second item
- Etc..

1. The first item in an ordered list
2. The second item
3. Etc..

#INCLUDE test_include.bml

// BEGIN_SRC #13 in README.org
/Here's/ *an* =example= (italic, bold, monospace)
