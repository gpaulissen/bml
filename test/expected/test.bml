#+TITLE: Precision club

#+DESCRIPTION: Strong club system. Nebulous diamond and 5 card majors

#VUL 00

#SEAT 0

1C = 3+ minor
1D = 3+ minor
1H = 5+ major
1S = 5+ major
1N = 15--17\nMay have 5M, 6m or 5-4 minors
  2C = Stayman
    2D = No 4 card major
      2H = 5+!h, 4!s, invitational
      2S = 5+!s, invitational
      3H = Smolen (5+ cards in other major)
      3S = Smolen (5+ cards in other major)
    2H = 4+ suit
    2S = 4+ suit
    2N = 4-4 majors, minimum
    3C = 4-4 majors, maximum
  2D = Transfer
    2H = Transfer accept
    3H = Super accept
  2H = Transfer
    2S = Transfer accept
    3S = Super accept
2C = Strong and forcing
  2D = Waiting
2D = Weak
  2N = Ogust
2H = Weak
  2N = Ogust
2S = Weak
  2N = Ogust
2N = 20--21

#VUL 00

#SEAT 0

(1N)
  D = Strength, ca 15+
  2C = At least 5-4 majors
    (D)
      P = 5+!c, suggestion to play
      R = Asking for better/longer major
      2D = 5+!d, suggestion to play
    (P)
      2D = Asking for better/longer major
  2D = A weak major or a strong minor
  2H = Constructive
  2S = Constructive
  2N = 5-5 minors
  3C = Preemptive
  3D = Preemptive
  3H = Preemptive
  3S = Preemptive
