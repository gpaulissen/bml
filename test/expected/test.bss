*00{Precision club}=NYYYYYYStrong club system. Nebulous diamond and 5 card majors
001N=NYYYYYY015--17\nMay have 5M, 6m or 5-4 minors
002C=NYYYYYY008Strong and forcing
002N=NYYYYYY020--21
001C=NYYYYYY0083+ minor
001D=NYYYYYY0083+ minor
001H=NYYYYYY0085+ major
001S=NYYYYYY0085+ major
002D=NYYYYYY008Weak
002H=NYYYYYY008Weak
002S=NYYYYYY008Weak
001NP2C=NYYYYYY008Stayman
001NP2CP2D=NYYYYYY008No 4 card major
001NP2CP2DP2H=NYYYYYY0085+!h, 4!s, invitational
001NP2CP2DP2S=NYYYYYY0085+!s, invitational
001NP2CP2DP3H=NYYYYYY008Smolen (5+ cards in other major)
001NP2CP2DP3S=NYYYYYY008Smolen (5+ cards in other major)
001NP2CP2N=NYYYYYY04-4 majors, minimum
001NP2CP3C=NYYYYYY0084-4 majors, maximum
001NP2CP2H=NYYYYYY0084+ suit
001NP2CP2S=NYYYYYY0084+ suit
002CP2D=NYYYYYY008Waiting
002DP2N=NYYYYYY0Ogust
002HP2N=NYYYYYY0Ogust
002SP2N=NYYYYYY0Ogust
*001N=NYYYYYY0
*001ND=NYYYYYY0Strength, ca 15+
*001N2C=NYYYYYY008At least 5-4 majors
*001N2CD=NYYYYYY0
*001N2CDP=NYYYYYY05+!c, suggestion to play
*001N2CDR=NYYYYYY0Asking for better/longer major
*001N2CD2D=NYYYYYY0085+!d, suggestion to play
*001N2CP=NYYYYYY0
*001N2CP2D=NYYYYYY008Asking for better/longer major
*001N2D=NYYYYYY008A weak major or a strong minor
*001N2N=NYYYYYY05-5 minors
*001N2H=NYYYYYY008Constructive
*001N2S=NYYYYYY008Constructive
*001N3C=NYYYYYY008Preemptive
*001N3D=NYYYYYY008Preemptive
*001N3H=NYYYYYY008Preemptive
*001N3S=NYYYYYY008Preemptive
001NP2D=NYYYYYY008Transfer
001NP2DP2H=NYYYYYY008Transfer accept
001NP2DP3H=NYYYYYY008Super accept
001NP2H=NYYYYYY008Transfer
001NP2HP2S=NYYYYYY008Transfer accept
001NP2HP3S=NYYYYYY008Super accept
