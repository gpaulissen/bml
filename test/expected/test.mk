# generated by BML
test.bml: test_include.bml
