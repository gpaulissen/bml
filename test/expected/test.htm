<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC '-//W3C//DTD XHTML 1.0 Strict//EN' 'http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd'>
<html lang="en" xml:lang="en" xmlns="http://www.w3.org/1999/xhtml">
 <head>
  <meta content="text/html; charset=utf-8" http-equiv="Content-Type"/>
  <title>
   Precision club
  </title>
  <style type="text/css">
   div.start {
    display: inline-block;
    vertical-align: top;
    width: 60px;
}
    
div.bidtable {
    width:700px;
    padding-bottom: 20px;
}    

div.bidtable ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
}

div.bidtable li ul
{
    list-style-type: none;
    margin: 0;
    padding: 0;
    margin-left: 60px;
}

div.tree {
    list-style-type: none;
    padding-bottom: 20px;
}

div.tree ul {
    list-style-type: none;
    /* vline.png */
    background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAAKAQMAAABPHKYJAAAAA1BMVEWIiIhYZW6zAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAB3RJTUUH1ggGExMZBky19AAAAAtJREFUCNdjYMAEAAAUAAHlhrBKAAAAAElFTkSuQmCC);
    background-size: 2px;
    background-repeat: repeat-y;
    margin: 0;
    padding: 0;
    margin-left: 10px;
}

div.tree li.node {
    margin: 0;
    padding: 0 12px;
    line-height: 20px;
    /* node.png */
    background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAgAAAAUAQMAAACK1e4oAAAABlBMVEUAAwCIiIgd2JB2AAAAAXRSTlMAQObYZgAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB9YIBhQIJYVaFGwAAAARSURBVAjXY2hgQIf/GTDFGgDSkwqATqpCHAAAAABJRU5ErkJggg==);
    background-size: 12px;
    background-repeat: no-repeat;
}

div.tree td.node {
    vertical-align: top;
    width: 60px;
}

div.tree li.last {
    /* lastnode.png */
    background-color: #fff;
    background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAgAAAAUAQMAAACK1e4oAAAABlBMVEUAAwCIiIgd2JB2AAAAAXRSTlMAQObYZgAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB9YIBhQIIhs+gc8AAAAQSURBVAjXY2hgQIf/GbAAAKCTBYBUjWvCAAAAAElFTkSuQmCC) ;
    background-size: 12px;
    backrgound-repeat: no-repeat;
}

div.tree li.root {
    background-color: #fff;
    background-image: none;
    backrgound-repeat: no-repeat;
}

.ccolor{color:Green}
.dcolor{color:Orange}
.hcolor{color:Red}
.scolor{color:Black}

@media print {
    * {
        -webkit-print-color-adjust: exact !important;   /* Chrome, Safari */
				color-adjust: exact !important;                 /* Firefox */
		}
    div.tree ul {
        background-size: 1px;
    }
}
  </style>
 </head>
 <body>
  <div class="bidtable">
   <ul>
    <li>
     <div class="start">
      &empty;
     </div>
     <ul>
      <li>
       <div class="start">
        1m
       </div>
       3+ minor
      </li>
      <li>
       <div class="start">
        1M
       </div>
       5+ major
      </li>
      <li>
       <div class="start">
        1NT
       </div>
       15&ndash;17
      </li>
      <li>
       <div class="start">
       </div>
       May have 5M, 6m or 5-4 minors
      </li>
      <li>
       <div class="start">
        2
        <span class="ccolor">
         &clubs;
        </span>
       </div>
       Strong and forcing
      </li>
      <li>
       <div class="start">
        2X
       </div>
       Weak
      </li>
      <li>
       <div class="start">
        2NT
       </div>
       20&ndash;21
      </li>
     </ul>
    </li>
   </ul>
  </div>
  <div class="bidtable">
   <ul>
    <li>
     <div class="start">
      1NT-2
      <span class="ccolor">
       &clubs;
      </span>
     </div>
     <ul>
      <li>
       <div class="start">
        2
        <span class="dcolor">
         &diams;
        </span>
       </div>
       No 4 card major
       <ul>
        <li>
         <div class="start">
          2
          <span class="hcolor">
           &hearts;
          </span>
         </div>
         5+
         <span class="hcolor">
          &hearts;
         </span>
         , 4
         <span class="scolor">
          &spades;
         </span>
         , invitational
        </li>
        <li>
         <div class="start">
         </div>
         &nbsp;
        </li>
        <li>
         <div class="start">
          2
          <span class="scolor">
           &spades;
          </span>
         </div>
         5+
         <span class="scolor">
          &spades;
         </span>
         , invitational
        </li>
        <li>
         <div class="start">
         </div>
         &nbsp;
        </li>
        <li>
         <div class="start">
          3
          <span class="hcolor">
           &hearts;
          </span>
          <span class="scolor">
           &spades;
          </span>
         </div>
         Smolen (5+ cards in other major)
        </li>
        <li>
         <div class="start">
         </div>
         &nbsp;
        </li>
       </ul>
      </li>
      <li>
       <div class="start">
        2
        <span class="hcolor">
         &hearts;
        </span>
        <span class="scolor">
         &spades;
        </span>
       </div>
       4+ suit
      </li>
      <li>
       <div class="start">
        2NT
       </div>
       4-4 majors, minimum
      </li>
      <li>
       <div class="start">
        3
        <span class="ccolor">
         &clubs;
        </span>
       </div>
       4-4 majors, maximum
      </li>
     </ul>
    </li>
   </ul>
  </div>
  <div class="bidtable">
   <ul>
    <li>
     <div class="start">
      &empty;
     </div>
     <ul>
      <li>
       <div class="start">
        2
        <span class="ccolor">
         &clubs;
        </span>
       </div>
       Strong, forcing
       <ul>
        <li>
         <div class="start">
          2
          <span class="dcolor">
           &diams;
          </span>
         </div>
         Waiting
        </li>
       </ul>
      </li>
      <li>
       <div class="start">
        2X
       </div>
       Weak
       <ul>
        <li>
         <div class="start">
          2NT
         </div>
         Ogust
        </li>
       </ul>
      </li>
     </ul>
    </li>
   </ul>
  </div>
  <div class="bidtable">
   <ul>
    <li>
     <div class="start">
      (1NT)&ndash;
     </div>
     <ul>
      <li>
       <div class="start">
        Dbl
       </div>
       Strength, ca 15+
      </li>
      <li>
       <div class="start">
        2
        <span class="ccolor">
         &clubs;
        </span>
       </div>
       At least 5-4 majors
       <ul>
        <li>
         <div class="start">
          (D)
         </div>
         <ul>
          <li>
           <div class="start">
            Pass
           </div>
           5+
           <span class="ccolor">
            &clubs;
           </span>
           , suggestion to play
          </li>
          <li>
           <div class="start">
            Rdbl
           </div>
           Asking for better/longer major
          </li>
          <li>
           <div class="start">
            2
            <span class="dcolor">
             &diams;
            </span>
           </div>
           5+
           <span class="dcolor">
            &diams;
           </span>
           , suggestion to play
          </li>
         </ul>
        </li>
        <li>
         <div class="start">
          (P)
         </div>
         <ul>
          <li>
           <div class="start">
            2
            <span class="dcolor">
             &diams;
            </span>
           </div>
           Asking for better/longer major
          </li>
         </ul>
        </li>
       </ul>
      </li>
      <li>
       <div class="start">
        2
        <span class="dcolor">
         &diams;
        </span>
       </div>
       A weak major or a strong minor
      </li>
      <li>
       <div class="start">
        2
        <span class="hcolor">
         &hearts;
        </span>
        <span class="scolor">
         &spades;
        </span>
       </div>
       Constructive
      </li>
      <li>
       <div class="start">
        2NT
       </div>
       5-5 minors
      </li>
      <li>
       <div class="start">
        3X
       </div>
       Preemptive
      </li>
     </ul>
    </li>
   </ul>
  </div>
  <div class="bidtable">
   <ul>
    <li>
     <div class="start">
      1NT&ndash;
     </div>
     <ul>
      <li>
       <div class="start">
        2
        <span class="ccolor">
         &clubs;
        </span>
       </div>
       Stayman
      </li>
      <li>
       <div class="start">
        2
        <span class="dcolor">
         &diams;
        </span>
       </div>
       Transfer
       <ul>
        <li>
         <div class="start">
          2
          <span class="hcolor">
           &hearts;
          </span>
         </div>
         Transfer accept
        </li>
        <li>
         <div class="start">
          3
          <span class="hcolor">
           &hearts;
          </span>
         </div>
         Super accept
        </li>
       </ul>
      </li>
      <li>
       <div class="start">
        2
        <span class="hcolor">
         &hearts;
        </span>
       </div>
       Transfer
       <ul>
        <li>
         <div class="start">
          2
          <span class="scolor">
           &spades;
          </span>
         </div>
         Transfer accept
        </li>
        <li>
         <div class="start">
          3
          <span class="scolor">
           &spades;
          </span>
         </div>
         Super accept
        </li>
       </ul>
      </li>
     </ul>
    </li>
   </ul>
  </div>
  <h1>
   The 1
   <span class="ccolor">
    &clubs;
   </span>
   opening
  </h1>
  <p>
   Opening 1
   <span class="ccolor">
    &clubs;
   </span>
   shows at least 16+ hcp, and is forcing. The
continuation is fairly natural.
  </p>
  <p>
   Some hands might be upgraded to 1
   <span class="ccolor">
    &clubs;
   </span>
   due to distribution, but
wildly distributional hands might also be downgraded to avoid
problems if the opponents preempt.
  </p>
  <h2>
   The 1
   <span class="dcolor">
    &diams;
   </span>
   negative
  </h2>
  <p>
   Responding 1
   <span class="dcolor">
    &diams;
   </span>
   shows a hand which doesn't have enough values to
establish a game force.
  </p>
  <ul>
   <li>
    The first item in an unordered list
   </li>
   <li>
    The second item
   </li>
   <li>
    Etc..
   </li>
  </ul>
  <ol>
   <li>
    The first item in an ordered list
   </li>
   <li>
    The second item
   </li>
   <li>
    Etc..
   </li>
  </ol>
  <p>
   <em>
    Here's
   </em>
   <strong>
    an
   </strong>
   <code>
    example
   </code>
   (italic, bold, monospace)
  </p>
 </body>
</html>